shared/src/main/java/gov/hhs/cdc/trustedintermediary/external/hapi/HapiHelper.java

```diff
@@ -52,17 +90,120 @@ public static void addMetaTag(
+        organizationReference.setResource(sendingFacility);
+        messageHeader.setSender(organizationReference);
+        bundle.addEntry(new Bundle.BundleEntryComponent().setResource(sendingFacility));
+    }
+
+    public static Organization createFacilityOrganization() {
+        Organization organization = new Organization();
+        String organizationId = UUID.randomUUID().toString();
+        organization.setId(organizationId);
+        return organization;
+    }
+
+    public static Identifier getSendingFacilityNamespace(Bundle bundle) {
+        Organization sendingFacility = getSendingFacility(bundle);
+        List<Identifier> identifiers = sendingFacility.getIdentifier();
+        return getHDNamespaceIdentifier(identifiers);
+    }
+
+    // MSH.5 - Receiving Application
+    public static MessageHeader.MessageDestinationComponent getReceivingApplication(Bundle bundle) {
+        MessageHeader messageHeader = getMessageHeader(bundle);
+        return messageHeader.getDestinationFirstRep();
+    }
+
+    public static void setReceivingApplication(
+            Bundle bundle, MessageHeader.MessageDestinationComponent receivingApplication) {
+        MessageHeader messageHeader = getMessageHeader(bundle);
+        messageHeader.setDestination(List.of(receivingApplication));
+    }
+
+    public static MessageHeader.MessageDestinationComponent createReceivingApplication() {
+        MessageHeader.MessageDestinationComponent destination =
+                new MessageHeader.MessageDestinationComponent();
+        destination.setId(UUID.randomUUID().toString());
+        return destination;
+    }
+
+    // MSH.6 - Receiving Facility
+    public static Organization getReceivingFacility(Bundle bundle) {
+        MessageHeader messageHeader = getMessageHeader(bundle);
+        return (Organization) messageHeader.getDestinationFirstRep().getReceiver().getResource();
+    }
+
+    public static void setReceivingFacility(Bundle bundle, Organization receivingFacility) {
+        MessageHeader messageHeader = getMessageHeader(bundle);
+        String organizationId = receivingFacility.getId();
+        Reference organizationReference = new Reference("Organization/" + organizationId);
+        organizationReference.setResource(receivingFacility);
+        MessageHeader.MessageDestinationComponent destination =
+                new MessageHeader.MessageDestinationComponent();
+        destination.setReceiver(organizationReference);
+        messageHeader.setDestination(List.of(destination));
+        bundle.addEntry(new Bundle.BundleEntryComponent().setResource(receivingFacility));
+    }
+
+    // PID.3 - Patient Identifier List
+    public static List<Identifier> getPatientIdentifierList(Bundle bundle) {
+        Patient patient = (Patient) resourceInBundle(bundle, Patient.class);
+        if (patient == null) {
+            throw new NoSuchElementException("Patient not found in the bundle");
```

Comment:
You throw `NoSuchElementException` in a couple of places in this class.  What are your thoughts about instead either returning `null` or an empty collection (for example, returning an empty `List` here)?  I haven't thought through this enough to have an opinion, but I'm curious what your thoughts are.

Reply:
Yeah, agreed. I actually was changing this in another PR. Initially I added it stop the flow but I think it's better to return null or empty collection

Reply:
I think if it's something that should structurally _always_ be there it makes sense to throw an exception (like if you have a resource without an ID for instance?), but if it's an optional field, then returning null or empty and letting the caller decide if that's bad makes sense

Reply:
Right. This is why it could go either way. In the context of the transformation that uses it, it's required. But since it's a helper function that could be used by other transformations that may not have that requirement, I think it's better to just return null

Reply:
The `getMSHMessageHeader` method still throws `NoSuchElementException`.  Is this on purpose?  Looks like that method is eventually called by our transformations and the exception isn't caught.

Reply:
Well, the message header is absolutely required so something would be very wrong if it's not there. There's another PR that will change how the exception handling is currently done but I'll check that it's handled correctly

Reply:
Removed throwing the exception and returning now null for consistency